File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size();
        int midpoint;
        while (l < r) {
            midpoint = (l + r) / 2;
            if (target == nums[midpoint]) {
                return midpoint;
            }
            else if (target < nums[midpoint]) {
                r = midpoint;
            }
            else {
                l = midpoint + 1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # naive way is to sort
        if len(nums) == 0: 
                return 0
        length = 1
        max_length = 1
        nums_sorted = sorted(nums)
        for i in range(len(nums_sorted) - 1):
            if (nums_sorted[i + 1] - nums_sorted[i]) == 1:
                length += 1
                if(length > max_length):
                    max_length = length
            elif (nums_sorted[i + 1]) == nums_sorted[i]:
                continue
            else:
                length = 1
        return max_length